tools/gsnapshot/Menu.cs

```diff
@@ -0,0 +1,116 @@
+/*
+   Copyright 2022 Google LLC
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+using System;
+using System.Collections.Generic;
+
+#nullable enable
+namespace GSnapshot
+{
+    class Menu
+    {
+        private List<string> options;
+        private List<int> ids;
+        private readonly String title;
+
+        public Menu(string title, Dictionary<int, string> options)
+        {
+            this.title = title;
+            this.options = new List<string>(options.Values);
+            this.ids = new List<int>(options.Keys);
+        }
+
+        public int? Show()
```

Comment:
That method is rather long, maybe you want to consider extracting the inner part of the while loop into a separate method to increase readability.

Reply:
Moved part to ShowOptions.